api/src/services/authentication.ts

```diff
@@ -409,17 +410,28 @@ export class AuthenticationService {
 			issuer: 'directus',
 		});
 
-		await this.knex('directus_sessions')
-			.update({
-				token: newRefreshToken,
-				expires: refreshTokenExpiration,
-			})
-			.where({ token: refreshToken });
+		if (options?.session) {
+			await this.updateStatefulSession(record, refreshToken, newRefreshToken, refreshTokenExpiration);
+		} else {
+			// original stateless token behavior
+			await this.knex('directus_sessions')
+				.update({
+					token: newRefreshToken,
+					expires: refreshTokenExpiration,
+				})
+				.where({ token: refreshToken });
+		}
 
 		if (record.user_id) {
 			await this.knex('directus_users').update({ last_access: new Date() }).where({ id: record.user_id });
 		}
 
+		// clear expired sessions for the current user
+		await this.knex('directus_sessions')
```

Comment:
In terms of the performance impact of this can we not remove this await? That should ensure this runs outside the request/response context and should not impact response time. We can add a `.catch` to ensure no unhandled rejections

Reply:
We can, however im not sure that is required considering we also await this on the `login` endpoint which cleans up all expired sessions (independent of which user is logging in) so the main delay should happen on login while this query cleans up the short "grace period" sessions created during the session itself only for the current user.